pkg/scale/varying_data_type.go

```diff
@@ -5,11 +5,13 @@ package scale
 
 import (
 	"fmt"
+	"strings"
 )
 
 // VaryingDataTypeValue is used to represent scale encodable types of an associated VaryingDataType
 type VaryingDataTypeValue interface {
 	Index() uint
+	String() string
```

Comment:
I don't think we need to enforce this for all `VaryingDataTypeValue` implementations.

Reply:
I would say I agree with this

Reply:
Good point, removed from interface and changed `VaryingDataType`'s String method to be:

```go
func (vdt VaryingDataType) String() string {
	if vdt.value == nil {
		return "VaryingDataType(nil)"
	}
	stringer, ok := vdt.value.(fmt.Stringer)
	if !ok {
		return fmt.Sprintf("VaryingDataType(%v)", vdt.value)
	}
	return stringer.String()
}
```

Does that work with you? 🤔